shared/libraries/native_streaming_protocol/src/server_session_handler.cpp

```diff
@@ -138,8 +146,19 @@ ReadTask ServerSessionHandler::readSignalSubscribe(const void* data, size_t size
         return createReadStopTask();
     }
 
-    if (signalSubscriptionHandler(signalNumericId, signalIdString, true, clientId))
-        sendSubscribingDone(signalNumericId);
+    try
+    {
+        const auto signal = findSignalHandler(signalIdString);
+        const auto hasAccess = hasUserAccessToSignal(signal);
+
+        if (hasAccess && signalSubscriptionHandler(signalNumericId, signal, true, clientId))
+            sendSubscribingDone(signalNumericId);
+    }
+    catch (const NativeStreamingProtocolException& e)
+    {
+        LOG_W("Protocol warning: {}", e.what());
+    }
+
```

Comment:
For me it seems simpler to place the access verifications at a higher level in `NativeStreamingServerHandler::handleSignalSubscription`, without needing to modify the `OnSignalSubscriptionCallback` signature or introduce an additional `OnFindSignalCallback`.